bluesky/run_engine.py

```diff
@@ -1713,11 +1719,17 @@ def _collect(self, msg):
                 self.log.debug("Emitted Event Descriptor with name %r "
                                "containing data keys %r (uid=%r)", stream_name,
                                data_keys.keys(), descriptor_uid)
-                self._descriptors[desc_key] = doc
+                self._descriptors[desc_key] = (objs_read, doc)
                 self._sequence_counters[desc_key] = count(1)
             else:
-                descriptor_uid = self._descriptors[desc_key]['uid']
-
+                objs_read, doc = self._descriptors[desc_key]
+                d_objs = frozenset(data_keys)
+                if d_objs != objs_read:
+                    raise RuntimeError("Miss matched objects read, "
```

Comment:
Mismatched the word, not Miss Matched the [children's clothing store](http://littlemissmatched.com/).